datafusion/sqllogictest/test_files/order.slt

```diff
@@ -891,3 +891,32 @@ select column2, column1 from foo ORDER BY column2 desc, column1 desc;
 # Cleanup
 statement ok
 drop table foo;
+
+# Casting from numeric to string types breaks the ordering
+statement ok
+CREATE EXTERNAL TABLE ordered_table (
+  a0 INTEGER,
+  a INTEGER,
+  b INTEGER,
+  c INTEGER,
+  d INTEGER
+)
+STORED AS CSV
+WITH HEADER ROW
+WITH ORDER (c ASC)
+LOCATION '../core/tests/data/window_2.csv';
+
+query T
```

Comment:
Can you also add a plan test that shows plan has correponding `SortExec`. Similarly, can you add a where cast is towards `BIGINT` and in the final plan `SortExec` is optimized away.